sts/internal/types.go

```diff
@@ -508,16 +570,34 @@ func (r *RequestSecurityToken) toString(c14n bool) string {
 		fmt.Sprintf(`<TokenType>%s</TokenType>`, r.TokenType),
 		fmt.Sprintf(`<RequestType>%s</RequestType>`, r.RequestType),
 		r.Lifetime.C14N(),
-		fmt.Sprintf(`<Renewing Allow="%t" OK="%t"></Renewing>`, r.Renewing.Allow, r.Renewing.OK),
-		fmt.Sprintf(`<Delegatable>%t</Delegatable>`, r.Delegatable),
-		actas,
-		fmt.Sprintf(`<KeyType>%s</KeyType>`, r.KeyType),
-		fmt.Sprintf(`<SignatureAlgorithm>%s</SignatureAlgorithm>`, r.SignatureAlgorithm),
-		fmt.Sprintf(`<UseKey Sig="%s"></UseKey>`, r.UseKey.Sig),
-		`</RequestSecurityToken>`,
 	}
 
-	return strings.Join(body, "")
+	if r.RenewTarget == nil {
+		body = append(body,
+			fmt.Sprintf(`<Renewing Allow="%t" OK="%t"></Renewing>`, r.Renewing.Allow, r.Renewing.OK),
```

Comment:
Hi @dougm,

Would a `bytes.Buffer` be more performant here? I'm not sure, but since you're essentially just joining the lines together anyway, writing to a buffer directly with `fmt.Fprintf` and then getting the string may end up being more efficient?

Reply:
It might be, but it isn't often that clients issue a token.  I would wait on optimizing for a separate PR if we think it's worthwhile, since the other existing C14N's should get the same treatment in that case.  I'd also prefer to put time into getting go-xmldsig + go-saml to work for us, rather than optimize the code in this file that I'd love to just throw away :)